bug(macros): #[url_patterns(mode = ws)] generates call to non-existent WebSocketRouter::with_namespace

## Bug Description

The `#[url_patterns(mode = ws)]` attribute macro (introduced in #3782) emits a wrapper that calls `__router.with_namespace(...)` on the function's return value. When the function returns a `WebSocketRouter`, this fails to compile because `WebSocketRouter` does not implement `with_namespace`. The shared `build_wrapper_and_assertion` helper (`crates/reinhardt-core/macros/src/url_patterns.rs`) emits the same wrapper for all four modes (server, client, unified, ws), but `with_namespace` is only available on `ServerRouter`, `ClientRouter`, and `UnifiedRouter`.

## Steps to Reproduce

```rust
use reinhardt::WebSocketRouter;
use reinhardt::url_patterns;
use crate::config::apps::InstalledApp;

#[url_patterns(InstalledApp::auth, mode = ws)]
pub fn ws_url_patterns() -> WebSocketRouter {
    WebSocketRouter::new()
}
```

Compile any crate consuming this:

```
error[E0599]: no method named `with_namespace` found for struct `WebSocketRouter` in the current scope
```

Reproduced against `kent8192/reinhardt-web@eb6107d5` (branch `main`).

## Expected Behavior

`#[url_patterns(mode = ws)]` should compile against an empty `WebSocketRouter::new()` (no consumers), generating only the `ws_url_resolvers` module without producing an invalid `with_namespace` call. Either:

- Add `WebSocketRouter::with_namespace(self, &str) -> Self` (parallel to `ServerRouter`), or
- Adjust `build_wrapper_and_assertion` to skip namespace-wrapping for `mode = ws` (since WebSocket route paths are absolute today and not namespaced via prefix).

## Actual Behavior

```
error[E0599]: no method named `with_namespace` found for struct `WebSocketRouter` in the current scope
   --> dashboard/src/apps/auth/ws_urls.rs:14:1
    |
14  | #[url_patterns(InstalledApp::auth, mode = ws)]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `WebSocketRouter`
```

## Environment

- Rust version: 1.94.0
- OS: macOS 25.3 / Linux x86_64 (Ubuntu 24.04 CI)
- reinhardt-web revision: `eb6107d5e19e2e06d83784fce42fde86a5259f94` (branch `main`)

## Additional Context

- Surfaced while migrating `kent8192/reinhardt-cloud` dashboard to the new typed `#[url_patterns]` API (cloud-side tracking issue forthcoming).
- All three apps that have no per-handler WebSocket consumers cannot satisfy `#[routes]`'s expectation of a `<app>::ws_urls::ws_url_resolvers::__for_each_ws_url_resolver!` symbol, because `mode = ws` itself does not compile on `WebSocketRouter`.
- Related upstream PRs: #3782 (introduced `mode = ws`), #3812 (fixed 5 other ws macro bugs).

--- a/crates/reinhardt-core/src/ws.rs
+++ b/crates/reinhardt-core/src/ws.rs
@@ -122,6 +122,8 @@ pub struct WebSocketRouter {
 	names: Arc<RwLock<HashMap<String, String>>>,
 	/// Build-time consumer registrations (added by `consumer()` builder).
 	pending_consumers: Vec<WebSocketRoute>,
+	/// Optional namespace (app label) for this router.
+	namespace: Option<String>,
 }
 
 impl WebSocketRouter {
@@ -131,7 +133,27 @@ impl WebSocketRouter {
 			routes: Arc::new(RwLock::new(HashMap::new())),
 			names: Arc::new(RwLock::new(HashMap::new())),
 			pending_consumers: Vec::new(),
+			namespace: None,
 		}
+	}
+
+	/// Set the namespace for this router.
+	///
+	/// Parallel to `ServerRouter::with_namespace`, emitted by
+	/// `#[url_patterns(mode = ws)]` to pass the `AppLabel::path(...)`.
+	/// WebSocket route paths are absolute today and are not rewritten
+	/// with this namespace; the value is stored for parity with other
+	/// routers and future use. See reinhardt-web#3829.
+	pub fn with_namespace(mut self, namespace: impl Into<String>) -> Self {
+		self.namespace = Some(namespace.into());
+		self
+	}
+
+	/// Returns the namespace set via [`with_namespace`], if any.
+	///
+	/// [`with_namespace`]: Self::with_namespace
+	pub fn namespace(&self) -> Option<&str> {
+		self.namespace.as_deref()
 	}
 
 	/// Register a WebSocket consumer by its factory function.
@@ -303,6 +325,18 @@ mod tests {
 		let route = router.find_pending("chat_ws");
 		assert!(route.is_some());
 		assert_eq!(route.unwrap().path(), "/ws/chat/{room_id}/");
+	}
+
+	#[rstest]
+	fn test_with_namespace_stores_value_without_rewriting_paths() {
+		let router = WebSocketRouter::new()
+			.with_namespace("auth")
+			.consumer(|| TestConsumer);
+		assert_eq!(router.namespace(), Some("auth"));
+		assert_eq!(
+			router.find_pending("chat_ws").unwrap().path(),
+			"/ws/chat/{room_id}/"
+		);
 	}
 
 	#[rstest]

